turtlebot3_example/turtlebot3_example/turtlebot3_obstacle_detection/turtlebot3_obstacle_detection.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+#
+# Copyright 2019 ROBOTIS CO., LTD.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors: Ryan Shim, Gilbert
+
+from geometry_msgs.msg import Twist
+import numpy
+from rclpy.node import Node
+from rclpy.qos import QoSProfile
+from rclpy.qos import qos_profile_sensor_data
+from sensor_msgs.msg import LaserScan
+
+
+class Turtlebot3ObstacleDetection(Node):
+
+    def __init__(self):
+        super().__init__('turtlebot3_obstacle_detection')
+
+        """************************************************************
+        ** Initialise variables
+        ************************************************************"""
+        self.linear_velocity = 0.0  # unit: m/s
+        self.angular_velocity = 0.0  # unit: m/s
+        self.scan_ranges = numpy.ones(360) * numpy.Infinity  # Scan resolution is 360
+
+        """************************************************************
+        ** Initialise ROS publishers and subscribers
+        ************************************************************"""
+        qos = QoSProfile(depth=10)
+
+        # Initialise publishers
+        self.cmd_vel_pub = self.create_publisher(Twist, 'cmd_vel', qos)
+
+        # Initialise subscribers
+        self.scan_sub = self.create_subscription(
+            LaserScan,
+            'scan',
+            self.scan_callback,
+            qos_profile=qos_profile_sensor_data)
+        self.raw_cmd_vel_sub = self.create_subscription(
+            Twist,
+            'raw_cmd_vel',
```

Comment:
How about changed this topic name to `cmd_vel_raw`?

Reply:
modified